src/server/health_check.go

```diff
@@ -0,0 +1,37 @@
+package server
+
+import "net/http"
+
+// HealthCheck defines a readiness check for a Webhook struct based on the presence of its TLS certificate and key.
+// It requires the whole webhook as parameter to be able to RLock on the certificate for the presence confirmation.
+type HealthCheck struct {
```

Comment:
Major: I would use a `http.HandleFunc` instead in order to simplify the code. We do not need a new struct for just serving a simple readiness probe check. You have an example [here](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/)

Minor: I would change the name in favour of a name that makes clear the intentions.

If it is a readiness probe check based on the present of TLS cert + key, you could use something like `TLSReadyReadinessProbe`.